### PR TITLE
Trigger msi build from Elastic Agent pipeline

### PR DESCRIPTION
**Pull request by jlind23**
## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Author's Checklist

- [ ]

## Related issues

- 

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: .buildkite/pipeline.elastic-agent-package.yml [hunk 1]
```diff
@@ -99,6 +99,16 @@ steps:
           .buildkite/scripts/steps/package.sh
         artifact_paths:
           - "build/distributions/**/*"
+  
+  - label: ":elastic-stack: Trigger Elastic Agent msi build"
+    trigger: "elastic-stack-installers"
+    build:
+      branch: main
```
Comment (cmacknz) [inline .buildkite/pipeline.elastic-agent-package.yml:106]: Do we intend to backport this? If we do the branch name needs to change correct? How will that be kept up to date as new release branches are created?

File: .buildkite/pipeline.elastic-agent-package.yml [hunk 2]
```diff
@@ -99,6 +99,16 @@ steps:
...
+      commit: HEAD
```
Comment (cmacknz) [inline .buildkite/pipeline.elastic-agent-package.yml:107]: This guarantees that the same commit is used in the MSI pipeline? That is, this is translated into commit hash XYZ which is what the triggered pipeline checks out? 

The triggered pipeline doesn't also check out HEAD? If it did this, there is a chance there was another commit to main between the two pipelines running.

File: .buildkite/pipeline.elastic-agent-package.yml (hunk 1 above)
Comment (jlind23) [reply .buildkite/pipeline.elastic-agent-package.yml:106]: This was just a draft PR to see if the trigger would work, the branch value will indeed be changed to rely on the current branch the pipeline is running on. I'll close this PR for now until the .msi PR is merged.

File: .buildkite/pipeline.elastic-agent-package.yml (hunk 2 above)
Comment (jlind23) [reply .buildkite/pipeline.elastic-agent-package.yml:107]: MSI always takes the latest snapshots available and use them to create the msi, they do not rely on a commit hash per say.

**Comment by jlind23** [comment]
Closing for now as it was just a test, will be reopened/reworked once the msi PR is merged.

**Pull request closed without merging**